metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -461,13 +461,13 @@ def run(self) -> None:
                         self.sink.write_record_async(record_envelope, callback)
 
                 self.sink.close()
-                self.process_commits()
                 self.final_status = "completed"
             except (SystemExit, RuntimeError, KeyboardInterrupt) as e:
                 self.final_status = "cancelled"
                 logger.error("Caught error", exc_info=e)
                 raise
             finally:
+                self.process_commits()
```

Comment:
does the impact other workflows that rely on stateful ingestion? e.g. time-window state for usage ingestion

I would like this behavior to change the behavior of stale entity removal, but not any of the other things

Reply:
Revert this change as there is no need and no use of moving it to finally block.